### DISCUSSION TITLE
What UI framework?

### DISCUSSION THREAD

**Discussion by justinmchase** [ayame113/ts-serve#26] [General]
I really like this approach, it really seems like the right way to be doing ESM in the browser.

But my next question is what UI framework would work well with this? Many big UI frameworks these days have their own compiler / bundler and webserver. Supposing I don't like any of that and would rather just use this, what UI framework works well with just focusing on the UI framework in the UI without bundling at all?

**Comment by ayame113** [comment]
Thank you for your interest in this library!

The use case for this library is when large frameworks are "overkill".
This library is meant to be just an extension of the static file server and is designed not to be tied to any particular UI framework.

Having said that, some UI library should work with this library.

- Preact / React
```tsx
// index.html
<script src="./main.ts" type="module"></script>
<div id="app"></div>

// main.ts
import { render } from 'https://esm.sh/preact@10.12.1';

function App() {
  return (
    <p class="big">Hello World!</p>
  )
}

render(<App />, document.getElementById("app"));
```
- WebComponent
  (Please also refer to https://simple-deno-website-boilerplate.onbrn.com/web-component)
```tsx
// index.html
<script src="main.ts" type="module"></script>
<my-element></my-element>

// main.ts
class MyElement extends HTMLElement {
  connectedCallback() {
    this.textContent = "hello";
  }
}
customElements.define("my-element", MyElement);
```

I think other libraries that are complete on the client side and distributed from a CDN will probably work as well.

**Reply by justinmchase** [reply]
Got it, so for example just plain react will work well but some other fully bundled system like `nextjs` may be too much?

**Reply by ayame113** [reply]
Yeah, I think it would be a separate project if you wanted something like next.js. (A framework like next.js should manage the routing and folder structure, but that's outside the scope of this project.)
